### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by andy128k**

### CODE REVIEW

File: src/memory.rs [hunk 1]
```diff
@@ -88,20 +79,28 @@ impl Memory {
         old_value
     }
 
-    pub fn set_bytes(&mut self, Address(start): Address, values: &[u8]) {
-        let start = start as usize;
+    pub fn set_bytes(&mut self, start: Address, values: &[u8]) {
+        let start = start.to_usize();
 
         // This panics if the range is invalid
```
Comment (mre) [inline src/memory.rs]: Nice one! Would this still panic if the range is invalid? Don't think so. In this case we could remove the warning.

File: src/memory.rs [hunk 2]
```diff
@@ -88,20 +79,28 @@ impl Memory {
...
+        (STACK_ADDRESS_LO..=STACK_ADDRESS_HI).contains(&address)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_memory_set_bytes() {
```
Comment (mre) [inline src/memory.rs]: Thanks for the test.

File: src/memory.rs [hunk 3]
```diff
@@ -88,20 +79,28 @@ impl Memory {
...
-        for (dest, src) in slice.iter_mut().zip(values.iter()) {
-            *dest = *src;
-        }
+        self.bytes[start..end].copy_from_slice(values);
     }
 
-    pub fn is_stack_address(address: &Address) -> bool {
-        STACK_ADDRESS_LO <= *address && *address <= STACK_ADDRESS_HI
+    pub fn is_stack_address(address: Address) -> bool {
+        (STACK_ADDRESS_LO..=STACK_ADDRESS_HI).contains(&address)
```
Comment (mre) [inline src/memory.rs]: Whoa!

File: src/memory.rs (hunk 1 above)
Comment (andy128k) [reply src/memory.rs]: I've added a test to ensure that it still panics.

Comment (mre) [reply src/memory.rs]: Thanks! I think panicking is the only sane thing we can do here. 👍

**Pull request merged**